src/handlers/rfqm_handlers.ts

```diff
@@ -0,0 +1,127 @@
+import { ConfigManager } from '../utils/config_manager';
+import { schemaUtils } from '../utils/schema_utils';
+
+const RFQM_INDICATIVE_QUOTE_REQUEST = new Counter({
+    name: 'rfqm_handler_indicative_quote_requested',
+    help: 'Request made to fetch rfqm indicative quote',
+    labelNames: ['apiKey'],
+});
+
+const RFQM_INDICATIVE_QUOTE_NOT_FOUND = new Counter({
+    name: 'rfqm_handler_indicative_quote_not_found',
+    help: 'Request to fetch rfqm indicative quote returned no quote',
+    labelNames: ['apiKey'],
+});
+
+const RFQM_INDICATIVE_QUOTE_ERROR = new Counter({
+    name: 'rfqm_handler_indicative_quote_error',
+    help: 'Request to fetch rfqm indicative quote resulted in error',
+    labelNames: ['apiKey'],
+});
+
+export class RfqmHandlers {
+    constructor(private readonly _rfqmService: RfqmService, private readonly _configManager: ConfigManager) {}
+
+    public async getIndicativeQuoteAsync(req: express.Request, res: express.Response): Promise<void> {
+        RFQM_INDICATIVE_QUOTE_REQUEST.labels(req.header('0x-api-key') || 'N/A').inc();
+        const params = this._parseFetchIndicativeQuoteParams(req);
+        let indicativeQuote;
+        try {
+            indicativeQuote = await this._rfqmService.fetchIndicativeQuoteAsync(params);
+        } catch (err) {
+            req.log.error(err, 'Encountered an error while fetching an rfqm indicative quote');
+            RFQM_INDICATIVE_QUOTE_ERROR.labels(req.header('0x-api-key') || 'N/A').inc();
+            throw new Error('Unexpected error encountered');
+        }
+
+        if (indicativeQuote === null) {
+            RFQM_INDICATIVE_QUOTE_NOT_FOUND.labels(req.header('0x-api-key') || 'N/A').inc();
+            throw new NotFoundError('Unable to retrieve a price');
+        }
+
+        res.status(HttpStatus.OK).send(indicativeQuote);
+    }
+
+    private _parseFetchIndicativeQuoteParams(req: express.Request): FetchIndicativeQuoteParams {
+        // HACK - reusing the validation for Swap Quote as the interface here is a subset
+        schemaUtils.validateSchema(req.query, schemas.swapQuoteRequestSchema as any);
+        const apiKey = req.header('0x-api-key');
+        if (apiKey === undefined) {
+            throw new InvalidAPIKeyError('Must access with an API key');
+        }
+
+        if (!this._configManager.getRfqmApiKeyWhitelist().has(apiKey)) {
+            throw new InvalidAPIKeyError('API key not authorized for RFQM access');
+        }
+
+        // Parse string params
+        const { takerAddress } = req.query;
+
+        // Parse tokens
+        const sellTokenRaw = req.query.sellToken as string;
+        const buyTokenRaw = req.query.buyToken as string;
+        validateNotNativeTokenOrThrow(sellTokenRaw, 'sellToken');
+        validateNotNativeTokenOrThrow(buyTokenRaw, 'buyToken');
+
+        const { tokenAddress: sellToken, decimals: sellTokenDecimals } = getTokenMetadataOrThrow(
+            sellTokenRaw,
```

Comment:
If the function throws here, what would it look like to the end user? Will it be a 5xx or a 4xx? Based on what I see, it would be a 5xx because it run through the "exception" return path.

Can we make sure that a 404 error is bubbled back to the user as 404?

Reply:
Ah, yes, can add another test for this one! Currently only have a test for WETH vs ETH, but not for a nonexistent token.